demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -442,16 +442,57 @@ static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext
  *
  * @return uint32_t The generated random number.
  */
-static int32_t generateRandomNumber();
+static uint32_t generateRandomNumber();
 
 /* Callbacks used to handle different events. */
 
+/**
+ * @brief The OTA agent has completed the update job or it is in
+ * self test mode. If it was accepted, we want to activate the new image.
+ * This typically means we should reset the device to run the new firmware.
+ * If now is not a good time to reset the device, it may be activated later
+ * by your user code. If the update was rejected, just return without doing
+ * anything and we'll wait for another job. If it reported that we should
+ * start test mode, normally we would perform some kind of system checks to
+ * make sure our new firmware does the basic things we think it should do
+ * but we'll just go ahead and set the image as accepted for demo purposes.
+ * The accept function varies depending on your platform. Refer to the OTA
+ * PAL implementation for your platform in aws_ota_pal.c to see what it
+ * does for you.
+ *
+ * @param[in] event Event from OTA lib of type OtaJobEvent_t.
+ * @return None.
+ */
 static void otaAppCallback( OtaJobEvent_t event,
                             void * pData );
+
+/**
+ * @brief Callback that notifies the OTA library when a job document is received.
```

Comment:
This says that the callback notifies the OTA library even though this callback is invoked _by_ the OTA library to notify the application so that the application can determine whether this is a JOB document or not and signal back to the OTA library accordingly?
If my understanding is correct, it would be good to mention that this is a "callback registered with the OTA library that notifies the OTA agent if an incoming PUBLISH message is a job document"

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/7bed0e4558dfd4a6c985fc6da34260501e76d4c7

Reply:
Reverted this change to be consistent with other demos.